Jenkinsfile

```diff
@@ -37,65 +37,90 @@ void unittests(String version) {
-                             alwaysLinkToLastBuild: true,
-                             keepAll     : true,
-                             reportDir   : 'test-reports/coverage/',
-                             reportFiles : 'index.html',
-                             reportName  : "Coverage${version}",
-                             reportTitles: "Coverage${version}"])
-                    }
-                    step([$class: 'JUnitResultArchiver', testResults: 'test-reports/junit/*.xml'])
-                }
-            }
-        }
-    }
+				}
+				catch(ex) {
+					throw ex
+				}
+				finally {
+					if (fileExists('test-reports/coverage/index.html')) {
+						publishHTML([allowMissing: true, alwaysLinkToLastBuild: true, keepAll: true, reportDir: 'test-reports/coverage/', reportFiles: 'index.html', reportName: "Coverage${version}", reportTitles: "Coverage${version}"])
+					}
+					step([$class: 'JUnitResultArchiver', testResults: 'test-reports/junit/*.xml'])
+				}
+			}
+		}
+	}
 }
 
 try {
-    Map tasks = [ : ]
-
-    pythonVersion.each {
-        version ->
-            tasks[version] = { unittests(version) }
-    }
-
-    tasks['snyk'] = {
-        stage('snyk') {
-            Snyk snyk = new Snyk(this, bparams)
-            snyk.execute()
-        }
-    }
-
-    tasks['sonarqube'] = {
-        stage('sonarqube') {
-            SonarQube.execute(this, bparams)
-        }
-    }
-
-    tasks['nexusiq'] = {
-        stage('nexusiq') {
-            Nexusiq.execute(this, bparams)
-        }
-    }
-
-    parallel(tasks)
-
-    common.setResultIfNotSet(Constants.JSUCCESS)
-} 
-catch (ex) {
-    common.logException(ex)
-    common.setResultAbortedOrFailure()
-    throw ex
-} 
-finally {
-    common.setResultIfNotSet(Constants.JFAILURE)
-    buildsCommon.notifyPostBuild(bparams)
+	Map tasks = [: ]
+
+	pythonVersion.each {
+		version ->tasks[version] = {
+			unittests(version)
+		}
+	}
+
+	tasks['snyk'] = {
+		stage('snyk') {
+			Snyk snyk = new Snyk(this, bparams)
+			snyk.execute()
+		}
+	}
+
+	tasks['sonarqube'] = {
+		stage('sonarqube') {
+			SonarQube.execute(this, bparams)
+		}
+	}
+
+	tasks['nexusiq'] = {
+		stage('nexusiq') {
+			Nexusiq.execute(this, bparams)
+		}
+	}
+
+	tasks['runPylint'] = {
+		stage('runPylint') {
+			node(Constants.DOCKERNODE) {
+				buildsCommon.cleanup()
+				checkout scm
+				
+				withContainer(image: "python:3.6-buster", registry: '', inside: '-u root') {
+					try {
+						sh """
+						mkdir reports
+						cd reports
+						touch pylint_tenable.log
+						touch pylint_tests.log				
+						cd ..
+						pip install pylint
+                           			pylint --exit-zero --output-format=parseable --reports=n tenable > reports/pylint_tenable.log
+                           			pylint --exit-zero --output-format=parseable --reports=n tests > reports/pylint_tests.log                           			
+                        			"""
+					} catch(ex) {
+						throw ex
+					} finally {
+						result = recordIssues(
```

Comment:
The plugin does not allow to have two pylint executions recorded.

java.lang.IllegalStateException: ID pylint is already used by another action: io.jenkins.plugins.analysis.core.model.ResultAction for Pylint

Please try `pyLint(id: "tenable", ....` to make the entries unique.

Reply:
See: https://www.jenkins.io/doc/pipeline/steps/warnings-ng/